### PR TITLE
Add ADMX definition for Group Policy

### PR DESCRIPTION
**Pull request by lechacon**
Adding definition file for a group policy. See #154 

Still pending:
* Setting up localization for the strings.
* Figuring out a representation for the lists of allowed and additional sources. Currently it is only a list of name/value pairs but we can't represent all three of name, arg & type.

Code to respect the policy will come as a separate PR

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/809)

### CODE REVIEW

**Comment by JohnMcPMS** [comment]
I think @denelon needs to approve this for location and content.

We will need to investigate how to integrate this with the localization build; I can help inform you of how the current one works.

Not really part of the PR, but I think that we do need to have a process for how we release and version this as well.  At a minimum, new policies will need documentation to indicate when they will be effective.

File: doc/admx/WindowsAppInstaller.admx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      <parentCategory ref="WindowsAppInstaller" />
+      <supportedOn ref="windows:SUPPORTED_Windows_10_0_RS3" />
+      <enabledValue>
+        <decimal value="1" />
+      </enabledValue>
+      <disabledValue>
+        <decimal value="0" />
+      </disabledValue>
+    </policy>
+    <policy name="EnableDefaultSorce" class="Machine" displayName="$(string.EnableDefaultSorce)" explainText="$(string.EnableDefaultSorceExplanation)" key="Software\Policies\Microsoft\Windows\WindowsPackageManager" valueName="EnableDefaultSorce">
```
Comment (JohnMcPMS) [inline doc/admx/WindowsAppInstaller.admx:66]: Typo in "Sorce"

File: src/AppInstallerCLI.sln [hunk 2]
```diff
@@ -473,10 +479,14 @@ Global
 		{952B513F-8A00-4D74-9271-925AFB3C6252} = {8D53D749-D51C-46F8-A162-9371AAA6C2E7}
 		{2ACDE176-F13F-42FA-8159-C34FA3D37837} = {8D53D749-D51C-46F8-A162-9371AAA6C2E7}
 		{866C3F06-636F-4BE8-BC24-5F86ECC606A1} = {60618CAC-2995-4DF9-9914-45C6FC02C995}
+<<<<<<< HEAD
```
Comment (yao-msft) [inline src/AppInstallerCLI.sln:482]: Leftover from merge?

File: src/AppInstallerCommonCore/GroupPolicy.cpp [hunk 3]
```diff
@@ -12,8 +12,7 @@ namespace AppInstaller::Settings
     {
         GroupPolicy& InstanceInternal(std::optional<GroupPolicy*> overridePolicy = {})
         {
-            // TODO: Read from the actual registry key
-            static GroupPolicy s_groupPolicy{ Registry::Key{} };
+            static GroupPolicy s_groupPolicy{ Registry::Key::OpenIfExists(HKEY_LOCAL_MACHINE, "Software\\Policies\\Microsoft\\Windows\\AppInstaller") };
```
Comment (yao-msft) [inline src/AppInstallerCommonCore/GroupPolicy.cpp:15]: >"Software\\Policies\\Microsoft\\Windows\\AppInstaller" [](start = 94, length = 54)

Should we make this const?

**Comment by lechacon** [comment]
Using final names/strings from discussion with @denelon
Here's a summary:
File names: DesktopAppInstaller.adm[x,l]
Location of the policies in gpedit: Computer Configuration\AdministrativeTemplates\Windows Components\Desktop App Installer
Location of the backing registry keys: HKLM:\Software\Policies\Microsoft\Windows\AppInstaller
Policy names use "App Installer", descriptions use "Windows Package Manager"

I set the supported OS to SUPPORTED_Windows_10_0_RS3 = version 1709.

File: src/AppInstallerCLI.sln (hunk 2 above)
Comment (lechacon) [reply src/AppInstallerCLI.sln]: Yes, it was. Fixed.

File: src/AppInstallerCommonCore/GroupPolicy.cpp (hunk 3 above)
Comment (lechacon) [reply src/AppInstallerCommonCore/GroupPolicy.cpp]: Done.

**Review by denelon — APPROVED** [review]
This looks good to me.

**Pull request merged**